src/classes/dataSource.cpp

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "classes/dataSource.h"
+#include "templates/optionalRef.h"
+
+// Return enum options for data source types
+EnumOptions<DataSource::DataSourceType> DataSource::dataSourceTypes()
+{
+    return EnumOptions<DataSource::DataSourceType>("DataSourceType", {{Internal, "Internal"}, {External, "External"}});
+}
+
+// Return data name (tag or filename)
+std::string_view DataSource::dataName() const { return dataName_; }
+
+// Return whether or not the data has been sourced and exists
+bool DataSource::dataExists() const
+{
+    return (!internalDataSource_.empty() && dataSourceType_ == Internal) ||
+           (!externalDataSource_.valueless_by_exception() && !data_.valueless_by_exception() && dataSourceType_ == External);
+}
+
+// Return data source type enum
+DataSource::DataSourceType DataSource::dataSourceType() const { return dataSourceType_; }
+
+// Return internal data source
+std::optional<std::string> DataSource::internalDataSource() const
+{
+    if (!internalDataSource_.empty())
+    {
+        return internalDataSource_;
+    }
+    return std::nullopt;
+}
+
+// Return external data source
+DataSource::Format &DataSource::externalDataSource() { return externalDataSource_; }
+
+// Function to source data (only required for internal data sources)
+bool DataSource::sourceData(GenericList &processingModuleData)
+{
+    if (!dataExists())
+    {
+        return false;
+    }
+    if (dataSourceType_ == Internal)
+    {
+        if (std::holds_alternative<Data1D>(data_) || std::holds_alternative<SampledData1D>(data_))
+        {
+            // Locate target data from tag and cast to base
+            auto optData = processingModuleData.searchBase<Data1DBase, Data1D, SampledData1D>(internalDataSource_);
+            if (!optData)
+            {
+                return Messenger::error("No data with tag '{}' exists.\n", internalDataSource_);
+            }
+            // Set data
+            data_ = optData->get();
+        }
+
+        // Data2D, Data3D
+        else
+        {
+            return std::visit(
+                [&]<class T>(T &data)
+                {
+                    // Locate target data from tag
+                    auto optData = processingModuleData.search<const T>(internalDataSource_);
```

Comment:
```suggestion
                [&](auto &data)
                {
                    // Locate target data from tag
                    auto optData = processingModuleData.search<const std::remove_reference<decltype(T)>>(internalDataSource_);
```

This removes the template on the lambda, which is illegal at the moment and breaking the Windows build.  You'll also need to `#include <type_traits` at the start of the file to use `std::remove:reference`.